From cpaasch@apple.com Thu Mar 31 02:40:48 2016
Return-path: <cpaasch@apple.com>
Envelope-to: baptiste@bitsofnetworks.org
Delivery-date: Thu, 31 Mar 2016 02:40:48 +0200
Received: from mail-out6.apple.com ([17.151.62.28] helo=mail-in6.apple.com)
	by archminux.polyno.me with esmtps (TLS1.0:DHE_RSA_AES_256_CBC_SHA1:256)
	(Exim 4.80)
	(envelope-from <cpaasch@apple.com>)
	id 1alQfH-0008Ag-Rn
	for baptiste@bitsofnetworks.org; Thu, 31 Mar 2016 02:40:48 +0200
DKIM-Signature: v=1; a=rsa-sha256; d=apple.com; s=mailout2048s; c=relaxed/simple;
	q=dns/txt; i=@apple.com; t=1459384838; x=2323298438;
	h=From:Sender:Reply-To:Subject:Date:Message-id:To:Cc:MIME-Version:Content-Type:
	Content-Transfer-Encoding:Content-ID:Content-Description:Resent-Date:Resent-From:
	Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:In-Reply-To:References:List-Id:
	List-Help:List-Unsubscribe:List-Subscribe:List-Post:List-Owner:List-Archive;
	bh=4wFvhO4ypu52V3PqW973+4UypKBTcuEPTkXkdUNGFtc=;
	b=bQ8oB14iTU6Nxkfyj/W8POFXGQdbb5S8ezKkZX1w7t6tw5NTtHxM5RRXA5ozWKSD
	kWzHVl+LNt/Mow55vB4J5WWPoIBHL0ecJCWW+7GpnYUNkMosfzaaLvsnn7NuKD7F
	jnOXbcJw7Xthi2WaBS9Bgzjttq1/Vvja1nqCI60YR6g+QZsup7SLe7cZ3FsPRaSb
	y80n/9zTR86yzhCq/kxEXCx3t1u+x22Cvrcr0/nF8l/oNMq1wyJy3UIq5U8JLOPC
	A5ndsjfG4FoN/1r1hmV+20wyAwsEem33Xj96ERq5AOB1zPeilKpMj1dDfpo6O4/d
	I+9cfuZnvgdDfqVZvuQ4Kw==;
Received: from relay6.apple.com (relay6.apple.com [17.128.113.90])
	by mail-in6.apple.com (Apple Secure Mail Relay) with SMTP id FA.8D.27179.6027CF65; Wed, 30 Mar 2016 17:40:38 -0700 (PDT)
X-AuditID: 11973e15-f79686d000006a2b-65-56fc72062a02
Received: from nwk-mmpp-sz07.apple.com (nwk-mmpp-sz07.apple.com [17.128.115.240])
	(using TLS with cipher DHE-RSA-AES128-SHA (128/128 bits))
	(Client did not present a certificate)
	by relay6.apple.com (Apple SCV relay) with SMTP id 8E.5B.18091.6027CF65; Wed, 30 Mar 2016 17:40:38 -0700 (PDT)
Received: from localhost ([17.149.220.228]) by nwk-mmpp-sz07.apple.com
 (Oracle Communications Messaging Server 7.0.5.35.0 64bit (built Mar 31 2015))
 with ESMTPSA id <0O4V00FAOOJQCTC0@nwk-mmpp-sz07.apple.com> for
 baptiste@bitsofnetworks.org; Wed, 30 Mar 2016 17:40:38 -0700 (PDT)
Sender: cpaasch@apple.com
From: Christoph Paasch <christoph.paasch@gmail.com>
To: mptcp-dev <mptcp-dev@listes.uclouvain.be>
Cc: Christoph Paasch <christoph.paasch@gmail.com>,
 Baptiste Jonglez <baptiste@bitsofnetworks.org>
Subject: [PATCH] mptcp: Do use tcp_v6_iif in mptcp_v6_do_rcv
Date: Wed, 30 Mar 2016 17:40:30 -0700
Message-id: <1459384830-773-1-git-send-email-christoph.paasch@gmail.com>
X-Mailer: git-send-email 2.6.3
X-Brightmail-Tracker: H4sIAAAAAAAAA+NgFrrHJMWRmVeSWpSXmKPExsUi2FAYpctW9CfMoO2apcWDs9NYHBg9+m+t
	ZwlgjOKySUnNySxLLdK3S+DK6J+lVrCcs+LB2Q7GBsYD7F2MnBwSAiYSZ5ovs0DYYhIX7q1n
	62Lk4hAS2Msosf/iYxaYoq7VS5kgEsuZJE5enwVVNYdJ4lfrJ7BRwgKSEt137jCD2GwCxhJP
	r8xgBbFFBHQl3r2fyAZiMwskS/zftJoNot5G4ti1L2A1LAKqEvOePADr5RXwkDhz6jwjxGY5
	iTlfFzKDLJMQWMYqcfr2UcYJjPwLGBlWMQrlJmbm6GbmmeklFhTkpOol5+duYgQFyHQ70R2M
	Z1ZZHWIU4GBU4uG9kPwnTIg1say4MvcQozQHi5I4L0sQUEggPbEkNTs1tSC1KL6oNCe1+BAj
	EwenVAOj5tK9/F/+6AvUih7r+Lbbkj/1tuB939mLnxx2cbsm/eO45O4fjAUv7/I/9WdIWlG/
	8NkV3jV31yjtzay/d6p5j3mJzeIWwb28f2YbXnPPk+R84ZLZ8Lz68ErdN0/fHlr50TfZJrrE
	KrKDd1qp9IZCj94+Ja559Xq5Ti9t83OT5s/fnvBiSsU8JZbijERDLeai4kQAxsOoqPEBAAA=
X-Brightmail-Tracker: H4sIAAAAAAAAA+NgFrrIJMWRmVeSWpSXmKPExsUi2FD8QZet6E+YweUZZhYPzk5jcWD06L+1
	niWAMYrLJiU1J7MstUjfLoEro3+WWsFyzooHZzsYGxgPsHcxcnJICJhIdK1eygRhi0lcuLee
	rYuRi0NIYDmTxMnrs6CcOUwSv1o/gXUIC0hKdN+5wwxiswkYSzy9MoMVxBYR0JV4934iG4jN
	LJAs8X/TajaIehuJY9e+gNWwCKhKzHvyAKyXV8BD4syp84wQm+Uk5nxdyDyBkWcBI8MqRoGi
	1JzESjO9xIKCnFS95PzcTYxgjxZG7WBsWG51iFGAg1GJh/dC8p8wIdbEsuLK3EOMEhzMSiK8
	/VFAId6UxMqq1KL8+KLSnNTiQ4zSHCxK4rzSWkApgfTEktTs1NSC1CKYLBMHp1QDo5z/5q5M
	7u0XglRcL828+0bwnLT1tNRd99ZFlGtUCryaNSc9f1H/TN9tkZwXWX86Z0jIaczcVfj6YSKD
	ymbn+LJut9XbVJPX3qq72RA02TX+V3tmwY2ptfqrwoM1UzY7nIo7Uruw7kCHmsGCuxe2ftr4
	r1pu41klkYwfgZnmZT/amnY8MXP7p8RSnJFoqMVcVJwIAPR2HkLkAQAA
Status: RO
X-Status: F
Content-Length: 1155
Lines: 36

TCP moves the IP6CB into a different region when in skb->cb[]. Thus, we
have to use tcp_v6_iif in appropriate places.

Cc: Baptiste Jonglez <baptiste@bitsofnetworks.org>
Fixes: 1b8a2b899837f (Merge tag 'v4.0' into mptcp_trunk)
Reported-by: Baptiste Jonglez <baptiste@bitsofnetworks.org>
Signed-off-by: Christoph Paasch <christoph.paasch@gmail.com>
---
 net/mptcp/mptcp_ipv6.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/mptcp/mptcp_ipv6.c b/net/mptcp/mptcp_ipv6.c
index bd5701539949..9daa69964f92 100644
--- a/net/mptcp/mptcp_ipv6.c
+++ b/net/mptcp/mptcp_ipv6.c
@@ -208,7 +208,7 @@ int mptcp_v6_do_rcv(struct sock *meta_sk, struct sk_buff *skb)
 						&tcp_hashinfo,
 						&ip6h->saddr, th->source,
 						&ip6h->daddr, ntohs(th->dest),
-						inet6_iif(skb));
+						tcp_v6_iif(skb));
 
 		if (!sk) {
 			kfree_skb(skb);
@@ -284,7 +284,7 @@ reset_and_discard:
 		 */
 		req = inet6_csk_search_req(meta_sk, th->source,
 					   &ipv6_hdr(skb)->saddr,
-					   &ipv6_hdr(skb)->daddr, inet6_iif(skb));
+					   &ipv6_hdr(skb)->daddr, tcp_v6_iif(skb));
 
 		if (req) {
 			inet_csk_reqsk_queue_drop(meta_sk, req);
-- 
2.6.3


